### ISSUE TITLE
BinnedSpikeTrain creates new spikes

### ISSUE THREAD

**Issue by pietromarchesi** [NeuralEnsemble/elephant#202]
I have spike times defined in microseconds. When I use BinnedSpikeTrain with a bin size defined in milliseconds, I end up with more spikes than I started with.

```python
binsize_in_ms = 20
spiketrains = []
for k in range(time_stamps.shape[0]):
    spike_times = time_stamps[k]
    train = neo.SpikeTrain(times=spike_times*pq.us, t_start=session_t_start,
                           t_stop=session_t_stop)
    spiketrains.append(train)


bs = elephant.conversion.BinnedSpikeTrain(spiketrains,
                                          binsize=binsize_in_ms * pq.ms,
                                          t_start=session_t_start,
                                          t_stop=session_t_stop)
print(np.sum([t.times.__len__() for t in spiketrains]))
print(bs.to_array().sum())
```
Output:
```
696394
696389
```

If I rescale my spike trains, I get another error:

```python
binsize_in_ms = 20

spiketrains = []
for k in range(time_stamps.shape[0]):
    spike_times = time_stamps[k]
    train = neo.SpikeTrain(times=spike_times*pq.us, t_start=session_t_start,
                           t_stop=session_t_stop)
    train = train.rescale(pq.ms)
    spiketrains.append(train)

bs = elephant.conversion.BinnedSpikeTrain(spiketrains,
                                          binsize=binsize_in_ms * pq.ms,
                                          t_start=session_t_start,
                                          t_stop=session_t_stop)
```
```
ValueError: some spike trains are not defined in the time given by t_start
```

Actually, it does not have to do with the units, it seems, but with the size of the binning window. This in fact gives the same problem:

```python
binsize_in_ms = 20
binsize_in_us = binsize_in_ms * 1000
# generate spike trains
spiketrains = []
for k in range(time_stamps.shape[0]):
    spike_times = time_stamps[k]
    train = neo.SpikeTrain(times=spike_times*pq.us, t_start=session_t_start,
                           t_stop=session_t_stop)
    spiketrains.append(train)


bs = elephant.conversion.BinnedSpikeTrain(spiketrains,
                                          binsize=binsize_in_us * pq.us,
                                          t_start=session_t_start,
                                          t_stop=session_t_stop)
print(np.sum([t.times.__len__() for t in spiketrains]))
print(bs.to_array().sum())
```
Where this does not:
```python
binsize_in_ms = 2
binsize_in_us = binsize_in_ms * 1000
```

**Comment by dizcza** [comment]
Could you please provide your `spike_times`, `session_t_start`, and `session_t_stop`? I can't reproduce this with simulated data

```python
import numpy as np
import quantities as pq

import elephant
from elephant.conversion import BinnedSpikeTrain
from elephant.spike_train_generation import homogeneous_gamma_process

binsize_in_ms = 2
spiketrains = [
    homogeneous_gamma_process(a=2.0, b=50 * pq.kHz, t_start=0 * pq.us,
                              t_stop=1e6 * pq.us)
    for i in range(10)]

binned_spikes = BinnedSpikeTrain(spiketrains, binsize=5 * pq.ms)
bs = elephant.conversion.BinnedSpikeTrain(spiketrains,
                                          binsize=binsize_in_ms * pq.ms,
                                          t_start=0 * pq.ms,
                                          t_stop=1000 * pq.ms)
print(np.sum([len(t.times) for t in spiketrains]))
print(bs.to_array().sum())
```

```
250376
250376
```

Print also please which elephant and neo versions you use.

**Comment by dizcza** [comment]
This issue should be fixed in #305 and #255 
If it is not, please reopen.
